### DISCUSSION TITLE
Return rewrites in getStatic/getServerProps

### DISCUSSION THREAD

**Discussion by sc-ruslanmatkovskyi** [vercel/next.js#57757] [Help]
### Summary

I have a host which has a list of sites where you can open [www.host.com/site1/](https://www.host.com/site1/), host.com/site2/ and etc. Each site has custom error pages 404 and 500 which are generated like static pages.
 
nextJS has a way which can return { redirect: { desitnation: 'redirectLink'}} in getStatic/ServerSideProps functions that use redirects when site is down
 
I need to find a solutions how to use rewrite it with the same method
 
it would be great if nextJS provides an option to return rewrites in getStatic/ServerSideProps like it works with redirect

For example:
    return {
      rewrite: {
        destination: resultErrorPages.notFoundPagePath,
        status: 404/500,
      }
}

**Comment by icyJoseph** [comment]
I think aside from the status... which you might still be able to hack in getServerSideProps...

Then you could always return `{props: {destination: 'something'}}` and use that to decide what to render within the actual page component. Somewhat similarly to how they do it here, https://nextjs.org/docs/pages/building-your-application/routing/custom-error#reusing-the-built-in-error-page:

```tsx
import Error from 'next/error'
 
export async function getServerSideProps() {
  const res = await fetch('https://api.github.com/repos/vercel/next.js')
  const errorCode = res.ok ? false : res.status
  const json = await res.json()
 
  return {
    props: { errorCode, stars: json.stargazers_count },
  }
}
 
export default function Page({ errorCode, stars }) {
  if (errorCode) {
    return <Error statusCode={errorCode} />
  }
 
  return <div>Next stars: {stars}</div>
}
```

Not for Errors of course, but to show other content under given circumstances. You'd need to code-split the alternatives to avoid having a massive bundle.
